js/main.js

```diff
@@ -0,0 +1,314 @@
+        return false     
+    };
+    calculationType(1)
+})
+
+buttons[7].addEventListener("click", function(e){      // +/-
+    e.preventDefault();
+    if (screen.value === "Infinity" || screen.value === "NaN" || screen.value === "undefined") {         
+        return false     
+    };
+
+    if (screen.value === "0") return false;
+
+    if (dropTheResult) {
+        dropTheResult = false;
+    }
+
+    screen.value.charAt(0) === "-" ? screen.value = screen.value.substring(1) : screen.value = "-" + screen.value;
+})   
+
+
+
+//     newNumber = true;
+//     calcType = null;    
+// })
+
+buttons[8].addEventListener("click", function(e){      //.
+    e.preventDefault();
+    if (screen.value === "Infinity" || screen.value === "NaN" || screen.value === "undefined") {         
+        return false     
+    };
+    
+    if (dropTheResult) {
+        result = null;
+        dropTheResult = false;
+    }
+
+    if (newNumber){
+        screen.value = "0"
+        newNumber = false
+    }
+
+    screen.value += "."
+})
+
+buttons[9].addEventListener("click", function(e){      // =
+    e.preventDefault();
+    if (screen.value === "Infinity" || screen.value === "NaN" || screen.value === "undefined") {         
+        return false     
+    };
+
+    if (screen.value === "") {                                      // if nothing is added
+        calcType = null;                                                // just in case some calculation button has already been pressed 
+        return false;
+
+    }   else if (result === null && calcType === null) {                // if no factor is inputted and calculation type is not specified
+        result = parseFloat(screen.value);
+
+    }   else if (result !== null && calcType === null) {                // if only one factor is inputted but calculation type is still not specified
+        screen.value = result;
+
+    }   else if (result !== null && calcType !== null && newNumber === true && factor2 === null) {       // if only one factor is inputted and calculation type is specified (but the second factor2 is not inputted) 
+        factor2 = result;
+        calculation();                                                                               
+        screen.value = result;
+
+    }   else if (result !== null && calcType !== null && newNumber === false && factor2 === null) {      // if only one factor is known, calcType is specified and "=" is pressed when there is some number in the screen.value                                                                             
+        factor2 = parseFloat(screen.value);
+        calculation();
+        screen.value = result;
+
+    }   else if (factor2 !== null) {       // if both factors and calculation type is specified                                                                             
+        calculation();
+        screen.value = result;
+    }  
+
+    dropTheResult = true;
+    newNumber = true;
+})
+
+
+
+// Digits 
+
```

Comment:
You could add only one listener on the table and handle digits event clicks by this listener:
```
const calculator = document.querySelector('.calculator');

calculator.addEventListener('click', e => {
  if (e.target.classList.contains('digitButton')) {
    const selectedNumber = e.target.innerText;
    
    if (screen.value === "Infinity" || screen.value === "NaN" || screen.value === "undefined") {        
      return false;
    };
    
    noNoughtStartsInteger();
    digit(selectedNumber);
  }
});
```